### DISCUSSION TITLE
ERR_REQUIRE_ESM when adding remark-frontmatter to Vite.js config

### DISCUSSION THREAD

**Discussion by msakrejda** [remarkjs/remark#870] [Q&A] [answered]
Hello,

I saw the other ERR_REQUIRE_ESM discussions here, but I wasn't able to figure out my problem based on those and I think my setup is slightly different so I thought I'd start a new discussion.

I'm trying to port a Rails/Webpacker project to Vite.js and I'm currently trying to get MD frontmatter parsing working in our MDX files. With webpacker, I had a simple loader that extracted it with `gray-matter`, but [vite-plugin-mdx](https://github.com/brillout/vite-plugin-mdx) docs mentioned remark-frontmatter so I figured I should try that.

When I added remark-frontmatter, I'm getting the following error when starting the Vite dev server:

```
09:34:06 vite-dev.1                | failed to load config from /home/maciek/project/vite.config.ts
09:34:06 vite-dev.1                | error when starting dev server:
09:34:06 vite-dev.1                | Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /home/maciek/project/node_modules/remark-frontmatter/index.js
09:34:06 vite-dev.1                | require() of ES modules is not supported.
09:34:06 vite-dev.1                | require() of /home/maciek/project/node_modules/remark-frontmatter/index.js from /home/maciek/project/vite.config.ts is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
09:34:06 vite-dev.1                | Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /home/maciek/project/node_modules/remark-frontmatter/package.json.
09:34:06 vite-dev.1                | 
09:34:06 vite-dev.1                |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
09:34:06 vite-dev.1                |     at Module.load (internal/modules/cjs/loader.js:928:32)
09:34:06 vite-dev.1                |     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
09:34:06 vite-dev.1                |     at Module.require (internal/modules/cjs/loader.js:952:19)
09:34:06 vite-dev.1                |     at require (internal/modules/cjs/helpers.js:88:18)
09:34:06 vite-dev.1                |     at Object.<anonymous> (/home/maciek/project/vite.config.ts:36:44)
09:34:06 vite-dev.1                |     at Module._compile (internal/modules/cjs/loader.js:1063:30)
09:34:06 vite-dev.1                |     at Object.require.extensions.<computed> [as .ts] (/home/maciek/project/node_modules/vite/dist/node/chunks/dep-36bf480c.js:77283:20)
09:34:06 vite-dev.1                |     at Module.load (internal/modules/cjs/loader.js:928:32)
09:34:06 vite-dev.1                |     at Function.Module._load (internal/modules/cjs/loader.js:769:14)
09:34:06 vite-dev.1                | exited with code 1
```

There are no requires in my code. I'm using remark-frontmatter 4.0.0, vite-plugin-mdx 3.5.6, typescript 4.3.2 and node 14.16. Everything worked fine before I added remark-frontmatter.

Any ideas?

**Comment by wooorm** [comment, marked as answer]
Probably vite-plugin-mdx related? You might be able to find more info on their issue tracker and/or have to downgrade this plugin.

**Reply by msakrejda** [reply]
Thanks. I looked there first, but no luck. I downgraded remark-frontmatter to 3.0.0 and that works around the issue (now I'm not seeing any changes, but that's probably a vite-plugin-mdx issue or a config issue on my part).

**Reply by obetomuniz** [reply]
I'm figuring the same issue. Still investigating the origin, but I am not using vite

**Reply by ChristianMurphy** [reply]
@obetomuniz see https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c for how to use ESM with a variety of build and bundle tools.
